README.md

```diff
@@ -23,6 +23,20 @@ See <https://community.flake.parts/services-flake/start>
 
 List of supported services is available at https://community.flake.parts/services-flake/services
 
+## Comparison with similar approaches
+
+| | services-flake | [devenv](https://devenv.sh/) |
+| --- | --- | --- |
+| macOS support | ✔️  | ✔️  |
+| Pure Flakes | ✔️  | ❌[^1]  |
+| Share services across flakes | ✔️[^2]  | ❌  |
+| Services as flake apps | ✔️  | ❌[^3]  |
```

Comment:
From https://devenv.sh/guides/using-with-flake-parts/

> If you're familiar with the Nix language and ecosystem, devenv **can be used without the devenv CLI** by integrating into [Nix Flakes](https://www.tweag.io/blog/2020-05-25-flakes/) using [flake-parts](https://flake.parts/).

But the page doesn't explain how exactly one starts devenv services without using its CLI.

Reply:
In the flake I created [here](https://github.com/juspay/services-flake/pull/135/files#r1517209132), I explored the flake in nix repl, and these are the packages it exports:

```sh
nix-repl> packages.x86_64-linux.
packages.x86_64-linux.container-processes  packages.x86_64-linux.default
packages.x86_64-linux.container-shell      packages.x86_64-linux.devenv-up
```

Reply:
But I can’t seem to be able to `nix run` it:

```sh
❯ nix run .#devenv-up --impure
warning: Git tree '/home/shivaraj/oss/devenc' is dirty
trace: warning: getExe: Package "cowsay-3.7.0" does not have the meta.mainProgram attribute. We'll assume that the main program has the same name for now, but this behavior is deprecated, because it leads to surprising errors when the assumption does not hold. If the package has a main program, please set `meta.mainProgram` in its definition to make this warning go away. Otherwise, if the package does not have a main program, or if you don't control its definition, use getExe' to specify the name to the program, such as lib.getExe' foo "bar".
error: unable to execute '/nix/store/67lf41ig3sf8z4i6lnigfn4c0icyb830-devenv-up/bin/devenv-up': Not a directory
```

Probably it doesn’t have a bin output

Reply:
This works fine: `❯ nix build .#devenv-up —impure`

And produces a result whose contents are:

```sh
❯ cat /nix/store/67lf41ig3sf8z4i6lnigfn4c0icyb830-devenv-up
#!/nix/store/087167dfxal194pm54cmcbbxsfy3cjgn-bash-5.2p26/bin/bash


/nix/store/lb2b16q8kc3xynj27d25sahhq95ph43h-process-compose-0.88.0/bin/process-compose --config /nix/store/vhxxz7j5bjlyi0ggfx4dlcf9an61p09q-process-compose.yaml \
  --port ${PC_HTTP_PORT:-9999} \
  --tui=${PC_TUI_ENABLED:-1} \
  up "$@" &


if [[ ! -d "$DEVENV_STATE" ]]; then
  mkdir -p "$DEVENV_STATE"
fi

stop_up() {
  echo "Stopping processes..."
  kill -TERM $(cat "$DEVENV_STATE/devenv.pid")
  rm "$DEVENV_STATE/devenv.pid"
  wait

  echo "Processes stopped."
}

trap stop_up SIGINT SIGTERM

echo $! > "$DEVENV_STATE/devenv.pid"

wait


```

Reply:
But when I try to run it, `postgres` service never starts, it just times out and fails with:

```sh
❯ ./result
mkdir: cannot create directory ‘’: No such file or directory
./result: line 25: /devenv.pid: Permission denied
```

Reply:
But the same configuration works (including `postgres` starting successfully) when I do `nix develop —impure` and then `devenv up`

Reply:
If I run `./result` after removing `process.implementation = “process-compose”`, it uses [honcho by default ](https://devenv.sh/reference/options/#processimplementation), which seems to work fine.

I was able to execute queries using `psql`:

```sh
shivaraj=# create table id(id int);
CREATE TABLE
shivaraj=# \dt
        List of relations
 Schema | Name | Type  |  Owner
--------+------+-------+----------
 public | id   | table | shivaraj
(1 row)
```

Reply:
> But when I try to run it, `postgres` service never starts, it just times out and fails with:
> 
> ```shell
> ❯ ./result
> mkdir: cannot create directory ‘’: No such file or directory
> ./result: line 25: /devenv.pid: Permission denied
> ```

This is because of not exporting `DEVENV_STATE`, because I am no longer in devShell, I will have to do it manually:

```sh
 DEVENV_STATE=$(pwd) ./result
```

But even after that I couldn’t get it running using the `process-compose` implementation without being in the devShell

Reply:
In conclusion, we can confidently say that devenv doesn't support running services (nevermind individual process groups) as a flake app.

Reply:
Looks like it, yes